client/ayon_core/tools/workfiles/widgets/files_widget_workarea.py

```diff
@@ -394,3 +406,28 @@ def _on_expected_selection_change(self, event):
         self._controller.expected_workfile_selected(
             event["folder"]["id"], event["task"]["name"], workfile_name
         )
+
+    def _on_model_refresh(self):
+        if (
+            not self._change_selection_on_refresh
+            or self._proxy_model.rowCount() < 1
+        ):
+            return
+
+        # Find the row with latest date modified
+        latest_index = max(
+            (self._proxy_model.index(i, 0) for
+             i in range(self._proxy_model.rowCount())),
+            key=lambda model_index: model_index.date(DATE_MODIFIED_ROLE)
```

Comment:
```suggestion
            (
                self._proxy_model.index(idx, 0) for
                idx in range(self._proxy_model.rowCount())
            ),
            key=lambda model_index: model_index.date(DATE_MODIFIED_ROLE)
```

Reply:
Question: Is last modified file the best option? Shouldn't be use the last of sorted by name?

Reply:
I'd say so yes - in some cases we copy paste a workfile from another project - as a starting point to work from. If we'd always go by name then it may occur that my version number of my current workfile will never be 'higher alphabetically' than the other filename in the folder. Which is hard to workaround.

Whereas if latest modified, then just saving would intentionally make something the latest.

Would be happy to hear if others disagree though.